### PR TITLE
CR: Fixed duplicate records for cr re-registration

### PR DESCRIPTION
**Pull request by peterwillcn**
Signed-off-by: Xiaobo Tian <peterwillcn@gmail.com>

### CODE REVIEW

File: cr/state/state.go [hunk 1]
```diff
@@ -280,15 +280,31 @@ func (s *State) registerCR(tx *types.Transaction, height uint32) {
...
+		s.history.Append(height, func() {
+			delete(s.Nicknames, c.Info().NickName)
+			delete(s.CanceledCandidates, c.Info().DID)
+			s.Nicknames[nickname] = struct{}{}
+			s.PendingCandidates[info.DID] = &candidate
+		}, func() {
+			s.Nicknames[nickname] = struct{}{}
+			s.PendingCandidates[info.DID] = &candidate
+			delete(s.Nicknames, c.Info().NickName)
+			delete(s.CanceledCandidates, c.Info().DID)
```
Comment (RainFallsSilent) [inline cr/state/state.go]: should change to :
delete(s.Nicknames, nickname)
delete(s.PendingCandidates, info.DID)
s.Nicknames[c.Info().NickName) = struct{}{}
s.CanceledCandidates[c.Info().DID] = c

File: cr/state/state.go [hunk 2]
```diff
@@ -280,15 +280,31 @@ func (s *State) registerCR(tx *types.Transaction, height uint32) {
...
+		})
+	}
+
```
Comment (RainFallsSilent) [inline cr/state/state.go]: returnDeposit should remove NickName from NickNames

File: cr/state/state.go [hunk 3]
```diff
@@ -280,15 +280,29 @@ func (s *State) registerCR(tx *types.Transaction, height uint32) {
...
+			delete(s.PendingCandidates, info.DID)
+		})
+	} else {
+		s.history.Append(height, func() {
+			delete(s.CanceledCandidates, c.Info().DID)
+			s.Nicknames[nickname] = struct{}{}
+			s.PendingCandidates[info.DID] = &candidate
+		}, func() {
+			delete(s.PendingCandidates, info.DID)
+			s.Nicknames[c.Info().NickName] = struct{}{}
```
Comment (RainFallsSilent) [inline cr/state/state.go]: need to  delete(s.Nicknames, info.NickName)

Comment (raindust) [reply cr/state/state.go]: yes, this should be a mistake

Comment (peterwillcn) [reply cr/state/state.go]: Fixed

**Pull request merged**